app/users/admin/users-list/users-list.component.html

```diff
@@ -1,19 +1,24 @@
 ﻿<div class="user-search">
-  <md-input-container class="grow-width">
-    <input mdInput type="text" placeholder="{{ 'search...' | translate }}" [(ngModel)]="query" />
+  <md-input-container>
+    <input mdInput placeholder="{{ 'search...' | translate }}" [(ngModel)]="query" (keyup.enter)="findUsers()">
   </md-input-container>
-  <md-select [(ngModel)]="selectedOption" placeholder="{{ 'search for' | translate }}">
-    <md-option *ngFor="let searchOption of options" [value]="searchOption">
-      {{ searchOption | lowercase | translate }}
-    </md-option>
-  </md-select>
-  <md-select [(ngModel)]="selectedRole" placeholder="{{ 'only show role' | translate }}">
+  <button md-icon-button *ngIf="showingSearchResults" (click)="resetSearch()">
+    <md-icon>close</md-icon>
+  </button>
+  <button md-icon-button [disabled]="query.trim().length < 1" (click)="findUsers()">
+    <md-icon>search</md-icon>
+  </button>
+
+  <md-select [(ngModel)]="selectedRole" (ngModelChange)="resetList()" placeholder="{{ 'only show role' | translate }}">
     <md-option *ngFor="let role of roles" [value]="role">
```

Comment:
There's no method `resetList()` - the UI is blocked after changing the role selection. The search according to the role only works together with the query search. Just selecting a role should work.